authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/validator/SAMLConfigurationValidator.java

```diff
@@ -0,0 +1,73 @@
+package com.synopsys.integration.alert.authentication.saml.validator;
+
+import com.synopsys.integration.alert.api.common.model.ValidationResponseModel;
+import com.synopsys.integration.alert.api.common.model.errors.AlertFieldStatus;
+import com.synopsys.integration.alert.api.common.model.errors.AlertFieldStatusMessages;
+import com.synopsys.integration.alert.authentication.saml.model.SAMLConfigModel;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+@Component
+public class SAMLConfigurationValidator {
+    public ValidationResponseModel validate(SAMLConfigModel model) {
+        Set<AlertFieldStatus> statuses = new HashSet<>();
+
+        // Perform validation on fields if enabled
+        if (model.getEnabled().orElse(false)) {
+            if (StringUtils.isBlank(model.getEntityId())) {
+                statuses.add(AlertFieldStatus.error("entityId", AlertFieldStatusMessages.REQUIRED_FIELD_MISSING));
+            }
+            if (StringUtils.isBlank(model.getEntityBaseUrl())) {
+                statuses.add(AlertFieldStatus.error("entityBaseUrl", AlertFieldStatusMessages.REQUIRED_FIELD_MISSING));
+            } else {
+                addErrorStatusIfInvalidUrl(model.getEntityBaseUrl(), "entityBaseUrl", statuses);
+            }
+
+            // Convert to empty optional for blank metadata url and file path
+            Optional<String> optionalFilteredMetadataUrl = model.getMetadataUrl().filter(StringUtils::isNotBlank);
+            Optional<String> optionalFilteredMetadataFilePath = model.getMetadataFilePath().filter(StringUtils::isNotBlank);
+            // One of url or filepath must exist
+            if (optionalFilteredMetadataUrl.isEmpty() && optionalFilteredMetadataFilePath.isEmpty()) {  // TODO: verify message
+                statuses.add(AlertFieldStatus.error("metadataUrl, metadataFilePath", "One of either fields is required."));
```

Comment:
I would change the message to be something more along the lines of "Either the metadata URL or metadata file path must be provided."

Reply:
Updated to use messages old saml validation used